mlx/io/load.h

```diff
@@ -36,77 +38,102 @@ class Writer {
 
 class FileReader : public Reader {
  public:
-  explicit FileReader(std::ifstream is)
-      : is_(std::move(is)), label_("stream") {}
   explicit FileReader(std::string file_path)
-      : is_(std::ifstream(file_path, std::ios::binary)),
-        label_(std::move(file_path)) {}
+      : fd_(open(file_path.c_str(), O_RDONLY)), label_(std::move(file_path)) {}
+
+  ~FileReader() {
```

Comment:
I think this should be virtual defined in the base class and also defined for the writer as well.

Reply:
Good catch, fixed.